jenkins/jenkins.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2015 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package jenkins
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+)
+
+const jenkinsURL = "http://192.168.1.10:8080"
+const jenkinsJobName = "gui_kbfs_dokan_1"
+
+func parseQueueNumber(locationString string) string {
+	countIndex := strings.Index(locationString, "queue/item/")
+	if countIndex == -1 {
+		log.Printf("bad location format: %s", locationString)
+		return ""
+	}
+	countIndex += len("queue/item/")
+	return strings.TrimRight(locationString[countIndex:], "/")
+}
+
+// StartBuild starts the build and returns the queue number as a string from Jenkins
+func StartBuild(clientRev string, kbfsRev string, JSONupdateFilename string) (string, error) {
+
+	u, err := url.Parse(jenkinsURL + "/job/" + jenkinsJobName + "/buildWithParameters")
+	urlValues := url.Values{}
+	if clientRev != "" {
+		urlValues.Add("ClientRevision", clientRev)
+	}
+	if kbfsRev != "" {
+		urlValues.Add("KFSRevision", kbfsRev)
+	}
+	if JSONupdateFilename != "" {
+		urlValues.Add("JSON_UPDATE_FILENAME", JSONupdateFilename)
```

Comment:
That's an arbitrary Jenkins parameter name. It works! I made Jenkins so that if you set this to the _test one it will push the build to s3 without waiting for manual promotion.